pkgs/tools/system/gkraken/default.nix

```diff
@@ -0,0 +1,79 @@
+{ python3Packages
+, lib
+, fetchFromGitLab
+, meson
+, pkg-config
+, glib
+, ninja
+, desktop-file-utils
+, i2c-tools
+, gobject-introspection
+, gtk3
+, libnotify
+, dbus
+, wrapGAppsHook
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "gkraken";
+  version = "1.1.6";
+
+  src = fetchFromGitLab {
+    owner = "leinardi";
+    repo = "gkraken";
+    rev = version;
+    sha256 = "085zz6m7c3xzsrvkw50gbbz8l9fmswxj2hjya2f52dvgs8daijdy";
+  };
+
+  format = "other";
+
+  postPatch = ''
+    patchShebangs .
+  '';
+
+  nativeBuildInputs = [
+    meson
+    pkg-config
+    glib
+    ninja
+    gtk3
+    desktop-file-utils
+    wrapGAppsHook
+  ];
+
+  buildInputs = [
+    gobject-introspection
+    glib
+    gtk3
+    libnotify
+    dbus
+  ];
+
+  propagatedBuildInputs = with python3Packages; [
+    pygobject3
+    peewee
+    rx
+    injector
+    liquidctl
+    pyxdg
+    requests
+    matplotlib
+    dbus-python
+  ];
+
+  postInstall = ''
+    mkdir -p $out/lib/udev/rules.d
+    udev_start=$(expr $(grep -n "_UDEV_RULE =" $src/gkraken/interactor/udev_interactor.py | cut -d':' -f1) + 1)
+    udev_end=$(expr $(grep -n "^'''" $src/gkraken/interactor/udev_interactor.py | cut -d':' -f1) - 1)
+    sed -n "''${udev_start},''${udev_end}p" $src/gkraken/interactor/udev_interactor.py > $out/lib/udev/rules.d/60-gkraken.rules
+  '';
```

Comment:
The udev files are [in a string inside one of the Python files](https://gitlab.com/leinardi/gkraken/-/blob/8c8e19c598fe3f299dcb95d382d78ac594ce37f6/gkraken/interactor/udev_interactor.py#L25-39). There's no option to print them, just to install/uninstall them under `/lib/udev/rules.d/60-gkraken.rules`. I didn't want to manually copy-paste them into the code & echo them out incase the rules ever change.

Please suggest better ideas for extracting them if you have any!

Reply:
Can we import the file and print out the variable?

Reply:
Found a way of doing that.